### ISSUE TITLE
Custom icons for categories that represent specific domains

### ISSUE THREAD

**Issue by jshorland** [ushahidi/platform#1825]
each domain requires its own icon. Where that is displayed is a mystery to me. Will need the brains of design.

**Comment by justinscherer** [comment]
This story requires a minor (I think) update to our "Post card" module and our checkbox modules.

**Post card module**
Example of the updated post card module in English
![Responses mobile.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/593714fffc98e900201ba2a5/125516c66e82c728ace21e0d46da9189338d8dba8afaff4eab5ff9982c046b313916ac217586f76ae5002e540b5e4ff8420d5912aeff8e28a6e03c70825d02bfd3394d68d47f40f2ecdd42e64bef62994597b4ba804dc4e13427ef8c8e483d1803199bb2.png)

We re-factor the meta data at the top of the card to display the icon, the name of the domain (earth, water, air), and it's custom status (see issue #1824)

We should remove the icons from the custom status indicators (even if we could find an icon that expressed "open for review" and "in progress" perfectly, it would probably be meaningless without a label)

Here's what that would look like in Arabic:

![Responses mobile arabic.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/593714fffc98e900201ba2a5/125516c66e82c728ace21e0d46da9189338d8dba8afaff4eab5ff9982c0626612806aa2832c9a138ad4d71040c1b5ca6535d5a10fdf8dd28f3b06a228c5b08ed876d5f7cc73443aeec9003ad43eb7edf10d8eff6df0ed2e22f27e1818e4b371707108efc5836a69c49da15.png)

These cards have the same format in the mobile and web clients, so appropriate changes should be made in both clients. 

**Input**
When they are creating the survey or filtering, we don't really have much of a choice other than jamming the icons between the checkbox and the label.

Here's a web client example:
![survey web client.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/593714fffc98e900201ba2a5/125516c66e82c728ace21e0d46fb818835879ae998ecbd03a751f9912752653f6b14b1247a80e03cfd1839545a0b4fad155b0a42aaab8f2ca5b66a708a540ae4836f0b66da2f53fde6d640e258f075c95d8fa4be8c46c1ec342ce5838848371c05188eed10.png)

Here's an Ionic example:
![custom checkboxes mobile.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/593714fffc98e900201ba2a5/125516c66e82c728ace21e0d46eb8189378d8ee98ce1ba40af5fff8c2c55677e2606aa2775c7af6aaf50740b0a0c0abe4b4a5a41a8f8887fa5e33a26d85a0ceed13b5f74ce3c18febad60da313f871d51bdaebe5c01982fa3737e48a854e3a170c1a81f85537a69a4ecd4abb.png)

Looking for feedback from @middle8media @dalezak
@Celestine22 @jshorland: Are there any other requirements for the custom status indicators and the inclusion of icons that I am forgetting?

**Comment by dalezak** [comment]
@rjmackay any idea how we'll achieve custom icons per category tag?

**Comment by justinscherer** [comment]
The icons we're using are here:
[GISCON Kuwait Icons.zip](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/593714fffc98e900201ba2a5/125516c66e82c728ace21e0d46cfbda900adade9a4fca842ad49b0bd2a4929606b48e13b628ae92dbc56395b4d5a4bf840590b47a9ffdf79a0b13f70dc5c0be5d0640d76922f0dbefad341eb5ee779c11ac7a4b6984fc1e23c21ee828d463d160c1f80f95827ed.zip)

Let me know if you need different formats

**Comment by dalezak** [comment]
Hmm, @justinscherer I'm not sure whether it is possible via the API to assign a custom icon to a category tag. Here's an example of the JSON for a single tag:
```
"id": 1,
"url": "https://mobileappqa.api.ushahidi.io/api/v3/tags/1",
"parent_id": null,
"tag": "Parent category 1",
"slug": "parent-category-1",
"type": "category",
"color": null,
"icon": "tag",
"description": "testing",
"priority": 99,
"created": "2017-05-31T16:32:29+00:00",
"role": null,
"children": [
    {
        "id": 2,
        "url": "https://mobileappqa.api.ushahidi.io/api/v3/tags/2"
    }
],
"allowed_privileges": [
    "read",
    "search"
]
```

There is a `"icon": "tag"` attribute, although don't believe there is a way to assign it in the backend?

The app is currently using Ionic standard icons http://ionicframework.com/docs/ionicons/, which render according to the target platform. So if the app is running on iOS vs Android, the icons will look different.

**Comment by justinscherer** [comment]
@dalezak My understanding is that the client has requested that we use their proprietary icons as part of the white label. Looking at the website they're about to release, the icons seem to be a big part of their re-brand. 

@Celestine22 Did I misunderstand the icon requirement?

**Comment by justinscherer** [comment]
Looking at the ionic icons...if we don't have to use the client's specific icons we can use these ones:
Air: `` <ion-icon ios="ios-partly-sunny" md="md-partly-sunny"></ion-icon>``
Land: ``  <ion-icon ios="ios-leaf" md="md-leaf"></ion-icon>``
Water: ``  <ion-icon ios="ios-water" md="md-water"></ion-icon>``

**Comment by dalezak** [comment]
Since it appears there is no current way in the backend to associate a custom icon per tag, we have three possible options:

1. refactor the backend to allow upload of a custom image or icon per category tag
2. use a hack for the web app and mobile app to display an icon based on the tag name
3. tell the client its not possible to have custom icons per category tag

@justinscherer @rjmackay @jshorland @Celestine22 thoughts on this?

**Comment by rjmackay** [comment]
@dalezak Correct, there's no UI for setting `icon` on the tag object, but I can do it manually. With only 3 categories its hardly worth building a configuration UI.

If we set `icon` to `water` `air` `land` could we map those to the clients icons (or ionic icons, whichever works)?

It might be sensible to err towards their proprietary icons, just so we can match those on the web ui too.

**Comment by dalezak** [comment]
Cool, that can work so something like the following could serve the purpose:

```
<ion-icon ios="ios-pricetag" md="md-pricetag" *ngIf="tag.icon != 'air' && tag.icon != 'land' && tag.icon != 'water'"></ion-icon>
<ion-icon ios="ios-partly-sunny" md="md-partly-sunny" *ngIf="tag.icon == 'air'"></ion-icon>
<ion-icon ios="ios-leaf" md="md-leaf" *ngIf="tag.icon == 'land'"></ion-icon>
<ion-icon ios="ios-water" md="md-water" *ngIf="tag.icon == 'water'"></ion-icon>
```

**Comment by rjmackay** [comment]
@Celestine22 can you confirm with the client if we need their branded icons or can use generic ones?

**Comment by jshorland** [comment]
@justinscherer I think you've covered it. Using their icons in branding is ideal. Double checking that the solution also includes custom status indicators including pending, open, in progress, finished? (I see in progress, I'm assuming it would just be the same but with the different statuses).

**Comment by Celestine22** [comment]
@rjmackay on it! @justinscherer this looks good, do you want to join the next call with client to talk through this?

**Comment by justinscherer** [comment]
cc @middle8media

**Comment by willdoran** [comment]
@rjmackay For the client are we going to pull the icons in from the PL or just store them in the client? Or even just include the svg directly in the templates.

**Comment by rjmackay** [comment]
> For the client are we going to pull the icons in from the PL or just store them in the client? Or even just include the svg directly in the templates.
 
I think any of those is fine. It'll depend if the client requires their branded icons or not. For now substitute the simplest thing you can.

**Comment by justinscherer** [comment]
@Celestine22 

> @justinscherer this looks good, do you want to join the next call with client to talk through this?>
I'm happy to jump on any calls where you think I'll be useful :-). 

Given the timelines and that app development is already done (ish), it makes more sense as a demo/progress update to me, because we're not really looking for feedback/changes. What do you think?

**Comment by Celestine22** [comment]
@justinscherer great, could you please compile this in a quick presentation so I can demo since we haven't pushed to production? Let me know how you best think we can work around this. Thanks.

**Comment by justinscherer** [comment]
@Celestine22 when are we doing this update? Ideally, @dalezak would just screenshare the working code from the emulator on his local. Alternatively, if Dale's not available, we can walk through a video screen capture from his local. I'd alway rather demo working code instead of static screens.

**Comment by dalezak** [comment]
I'm away today but back online Tuesday, can grab screenshots then of the app. Does the deployment have any responses yet? If not, the list of responses will be blank.

**Comment by dalezak** [comment]
I can also share a beta build tomorrow so others can grab screenshots from the current app.

**Comment by dalezak** [comment]
@willdoran @rjmackay did you decide on how the backend will handle the custom category icons?

**Comment by rjmackay** [comment]
It depends a little if they go with categories or surveys for domains, however right now I'm not handling them on the backend at all. The icons are simply shipped in the fork of the client and I check the name of the survey.

**Comment by rjmackay** [comment]
- [ ] Add icons to create post drop down on web client

**Comment by Celestine22** [comment]
- Client selected 3 surveys for the different domains.
- We will also need to update the icons for the different surveys on the map.

**Comment by rjmackay** [comment]
This issue was moved to ushahidi/platform-fix-our-environment#22
